### PR TITLE
feat: add another script executable, add program options

### PR DESCRIPTION
**Pull request by asalzburger**
This PR partly harmonizes the Executables in the `Script/TrackingPerformance` folder.
Currently one can run:
  * ActsExampleResidualAndPulls
  * ActsExamplePerigeeParameters

both steered by program options, the efficiency & fake-rate executable with also follow in a separate PR.

The naming of the plots is slightly refined and a quite option to produce plots from shell scripts (e.g. automatically from test runs in the CI is added).

all_pulls

all_residuals

### CODE REVIEW

**Comment by asalzburger** [comment]
@XiaocongAi - I only implemented the fitting option for the smoothed pulls (don't think it makes sense otherwise), for the residuals I think we should do some RMS estimate.

Concerning range estimation:
  * for the `boundParametersResolution` script I have access to the tree, so I have implemented an automated range-determination for the histograms borders
  * for the `perigeeParameterResolution` there are some fixed ranges, I suppose ?

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C [hunk 1]
```diff
@@ -24,37 +24,70 @@
...
-                          const std::string& saveAs = "") {
+/// Plot the bound parameter resoltuions
+///
+/// (loc1, phi, theta, q/p, t) at all track states from root file produced by
+/// the RootTrajectoryStatesWriter
+///
+/// @param inFile the input root file
+/// @param treeNAme the input tree name (default: 'trackstates)
+/// @param outFile the output root file
+/// @param pTypes the track parameter types (prd, flt, smt)
```
Comment (XiaocongAi) [inline Examples/Scripts/TrackingPerformance/boundParamResolution.C:42]: This needs some update, I guess.

**Review by XiaocongAi — APPROVED** [review]
Hi @asalzburger , it's really nice feature to have. Approved.

**Comment by XiaocongAi** [comment]
> * for the `perigeeParameterResolution` there are some fixed ranges, I suppose ?

Yes, the residual in the `track_parameters_fitter.root` are fixed, which is not very good. It's better tor read those variables from `tracksummary_fitter.root`, just like reading the residual for bound parameters from `trackstates_fitter.root`. However, the `tracksummary_fitter.root` does not include the branch for direct residual/pull (we have the truth the fitted params). I will open a PR to add the residual&pull. Then, we can get the variable residual range for the `perigeeParameterResolution`.

**Pull request merged**